xapian-core/tests/api_cluster.cc

```diff
@@ -0,0 +1,40 @@
+#include <config.h>
+
+#include "api_cluster.h"
+
+#include <xapian.h>
+
+#include "apitest.h"
+#include "testutils.h"
+
+#include <string.h>
+#include <cstring>
+
+using namespace std;
+
+DEFINE_TESTCASE(euclidian_distance, backend) {
+	Xapian::Database db(get_database("etext.txt"));
+    string query = "good";
+    Xapian::Stem stemmer("en");
+
+    Xapian::QueryParser qp;
+    qp.set_database(db);
+    qp.set_stemmer(stemmer);
+
+    Xapian::Enquire enq(db);
+    enq.set_query(qp.parse_query(query));
+    Xapian::MSet matches = enq.get_mset(0, 30);
+
+	Xapian::MSetDocumentSource docs(matches);
+	Xapian::TermListGroup tlg;
+	tlg.add_documents(docs);
+	Xapian::Document doc1 = matches[0].get_document();
+	Xapian::Document doc2 = matches[1].get_document();
+	Xapian::EuclidianDistance d;
+	
+	int sim = d.similarity(tlg, doc1.termlist_begin(), doc1.termlist_end(), doc2.termlist_begin(), doc2.termlist_end());
+
+	TEST(sim != 0);
```

Comment:
If you use a different test database (say `apitest_allterms`) then you can calculate exactly what the similarity should be using euclidian distance.

You probably also want a test that the similarity _is_ 0 between two identical documents.